actions/publish-github-pages/action.yml

```diff
@@ -1,10 +1,106 @@
-#  image: 'Dockerfile'
   using: 'composite'
   steps:
-    - run: ${{ github.action_path }}/entrypoint.sh
+    - id: checkout-repos
+      run: |
+           pip install GitPython pyyaml
+           python -c 'from checkout_and_collate import *; checkout_repos()'
       shell: bash
-#    - run: ./actions/publish-github-pages/entrypoint.sh
+      env:
+        PYTHONPATH: ${{ github.action_path }}
+        GITHUB_PAGES_PUSH_PATH: ${{ github.action_path }}/${{ env.GITHUB_PAGES_DIRECTORY }}
+        INPUT_GITHUB_PAGES_PULL_BRANCH: ${{ inputs.GITHUB_PAGES_PULL_BRANCH }}
+        INPUT_GITHUB_PAGES_HUGO_PATH: ${{ inputs.GITHUB_PAGES_HUGO_PATH }}
+        INPUT_DEBUG: ${{ inputs.DEBUG }}
+        INPUT_STAGING_DIR: ${{ inputs.STAGING_DIR }}
+    - id: insert-custom-css
+      run: |
+           export CSS=${CSS:-$INPUT_CSS}
+           export GITHUB_PAGES_HUGO_PATH=${GITHUB_PAGES_HUGO_PATH:-$INPUT_GITHUB_PAGES_HUGO_PATH}
+           echo -e "$CSS" >> ${GITHUB_PAGES_HUGO_PATH}/themes/cloudposse/src/scss/custom.scss
+      shell: bash
+      env:
+        INPUT_CSS: ${{ inputs.CSS }}
+        INPUT_GITHUB_PAGES_HUGO_PATH: ${{ inputs.GITHUB_PAGES_HUGO_PATH }}
+    - id: collate-files
+      run: |
+           pip install GitPython pyyaml
+           python -c 'from checkout_and_collate import *; collate_files()'
+      shell: bash
+      env:
+        PYTHONPATH: ${{ github.action_path }}
+        GITHUB_PAGES_PUSH_PATH: ${{ github.action_path }}/${{ env.GITHUB_PAGES_DIRECTORY }}
+        INPUT_GITHUB_PAGES_HUGO_PATH: ${{ inputs.GITHUB_PAGES_HUGO_PATH }}
+        INPUT_GITHUB_PAGES_PULL_BRANCH: ${{ inputs.GITHUB_PAGES_PULL_BRANCH }}
+        INPUT_DEBUG: ${{ inputs.DEBUG }}
+        INPUT_CONTENT: ${{ inputs.CONTENT }}
+        INPUT_STAGING_DIR: ${{ inputs.STAGING_DIR }}
+    - id: build-hugo
+      run: |
+           # These variables need to be defined manually when GitHub Actions variables are passed in using
+           # with: instead of env: in order for the next steps to work.
+           export HUGO_URL=${HUGO_URL:-$INPUT_HUGO_URL}
+           export HUGO_CONFIG=${HUGO_CONFIG:-$INPUT_HUGO_CONFIG}
+           export HTMLTEST_CONFIG=${HTMLTEST_CONFIG:-$INPUT_HTMLTEST_CONFIG}
+           export HUGO_PUBLISH_DIR=${HUGO_PUBLISH_DIR:-$INPUT_HUGO_PUBLISH_DIR}
+           export STAGING_DIR=${STAGING_DIR:-$INPUT_STAGING_DIR}
+           cd $STAGING_DIR
+           docker build -t cloudposse/docs .
+           make release
+           make real-clean hugo/build
+      shell: bash
+      env:
+        INPUT_STAGING_DIR: ${{ inputs.STAGING_DIR }}
+        INPUT_HUGO_URL: ${{ inputs.HUGO_URL }}
+        INPUT_HUGO_PUBLISH_DIR: ${{ inputs.HUGO_PUBLISH_DIR }}
+        INPUT_HUGO_CONFIG: ${{ inputs.HUGO_CONFIG }}
+        INPUT_HTMLTEST_CONFIG: ${{ inputs.HTMLTEST_CONFIG }}
+    - id: publish-site
+      run: |
+           export STAGING_DIR=${STAGING_DIR:-$INPUT_STAGING_DIR}
+           export HUGO_PUBLISH_DIR=${HUGO_PUBLISH_DIR:-$INPUT_HUGO_PUBLISH_DIR}
```

Comment:
Can't we move this to `env`?

Reply:
I know it's ugly, but see my comment above about `INPUT_` variables. This is my way of accommodating both the `with:` and `env:` modes of passing in variables. If there's a better way of doing that, I'm open to it.